### DISCUSSION TITLE
1차 데모 회의

### DISCUSSION THREAD

**Discussion by jaehee329** [woowacourse-teams/2023-haru-study#14] [아이디어]
5분간 자기가 생각하는 프로젝트 방향성,
현재 상황에 맞는 일정 추정을 포함하여

~6:10

**Comment by yeopto** [comment]
1. 효능감을 줄 수 있는 기능이 무엇일지 고민하여 정한다.
2. 기존에 정했던 MVP 기능 명세에서 뺄 수 있는 것들 뺀다.
3. 그리고 MVP를 구현하고 2차 데모에 방향성을 제시한다.
4. 효능감을 줄 수 있는 기능을 구현한다.

**Reply by MoonJeWoong** [reply]
빠르고 짧은 스터디의 효능감을 위해 현재 준이 제시한 템플릿 말고도 다른 템플릿이나 자료를 조사하는 과정에 리소스를 투자하자는 의미로 해석하면 되겠지??

**Reply by woo-jk** [reply]
구구가 말했던 것 처럼 2차 데모까지 워킹데이가 대략 8일? 정도 남았는데 현실적으로 MVP 구현은 힘들 것 같다고 생각 ㅠ

3차 데모까지도 MVP 구현이 가능할지 확신을 할 수 없을 것 같은데, 만약 모집 플랫폼 구현이 예상보다 많이 길어지면 어떻게 해야할까?

**Comment by jaehee329** [comment]
### 2차 데모까지 모집 관련 기능에만 집중한다.
1. 뱃지 및 프로필 관련 기능을 제외한다
2. 로그인 및 모집 글의 CRUD만을 남긴다.
3. 스터디 룸은 옵션

### 2차 데모 이후에는 진행 관련 기능을 추가한다
- 완전 별개의 탭으로 스터디 도중에 사용할 수 있는 툴을 제공한다.
<img width="516" alt="image" src="https://github.com/woowacourse-teams/2023-haru-study/assets/77962265/80304d6d-d31c-4079-8e16-d88c598aa294">

인프런은 별개의 기능은 위와 같이 탭을 나눠 제공한다.
스터디 모집 및 프로필 기능은 기존처럼, 스터디 도중 사용할 수 있는 툴을 신규 탭에서 제공한다.
기존처럼 오픈 카카오톡 사용 정책은 유지하나, 스터디 도중 이 툴을 사용할 수 있도록 스터디 시작 시 툴 사용을 제안하는 등의 기능을 넣는다.


- 툴의 형태
준의 스터디 방법을 인용한다면, 여러 사용자들이 한 세션에서 사용할 수 있는 타이머를 제공하면 좋을 듯.

**Reply by aak2075** [reply]
뱃지 및 프로필 관련 제외는 좋아요👍

툴의 종류도 되게 다양한데 ex)줌, 피그젬, 채팅방 
이런것들중 어떤 종류가 사용자들이 정말 필요한지 우리가 파악하고 구현할 수 있을까?

**Reply by woosung1223** [reply]
오픈 카카오톡을 사용하면서, 툴은 우리 플랫폼 것을 사용하고 화상회의는 줌으로 한다면 유저가 관리해야 할 포인트가 너무 많아지지 않을까요? 
개인적으로는 메인 피쳐로 툴을 제공할거라면 오픈 카카오톡은 배제해야 한다고 생각하는데 어떻게 생각하세요?

**Reply by yeopto** [reply]
뱃지 및 프로필 관련 제외는 좋습니다.

스터디 룸은 꼭 필요하다고 생각해요. 룸이 없다면 사실 모집 기능이 필요없다고 생각해요.
모집을 하여 모집된 사람들이 룸에서 스터디하며 효능성을 줄 수 있는 방안이 우리가 기획한 부분에서 틀어지는게 많이 없을 것 같단 생각이에요.

**Reply by woo-jk** [reply]
@aak2075 
준이 제시한 스터디 방법을 우리가 직접 해봐서 검증해보면 어떨까? 우리가 사용하면서 만족스러운지, 만족스럽지 않은지 먼저 검증해보면 될 듯!

**Reply by MoonJeWoong** [reply]
툴의 기능을 너무 거창하게 잡지 말고, 준이 제시한 템플릿 같은 양식에 맞춰 내용을 작성할 수 있는 최소한의 text입력 기능을 최우선으로 잡고 진행하는 방향으로 하면 좋을 것 같다. 그리고 공유는 해당 스터디 상세 페이지에 댓글 같은 느낌으로 각자가 작성한 폼을 열어서 볼 수 있는 방식으로 하면 어떨까? (이 부분에서 나중에 스프레드 시트 등을 연결하거나 여기로 이동해서 공유 문서 작성 API를 연동하는 방향으로 가봐도 좋을 것 같다.)

**Reply by nlom0218** [reply]
완전 별개인거 좋은 거 같다. 굳이 스터디룸에서 진행할 필요는 없다고 생각이 드네. 스터디장이 상황에 맞게 편하게 쓸 수 있으면 좋을 듯!

스터디 생성 -> 스터디 진행 -> 진행 도중 툴 선택 -> 해당 툴이 적용될 스터디 선택(유저 초대) -> 스터디룸에서 스터디 툴이 보임: 이런 플로우라면 별개로 개발을 진행하고 나중에 붙이기 좋을 거 같담

**Comment by aak2075** [comment]
핵심 가치를 제공하는데 집중해야 할 것 같다.

1. 프로필, 뱃지 기능은 우선순위에서 제외하고 나중에 추가할지 고려해보자.
2. 준이 제시한 방법이나 짧은 주제를 효과적으로 학습할 수 있는 방법을 논문이나 자료를 찾아서 스터디 개설 폼에서 이를 입력받도록 한다.

진행을 도와준다기 보다는 모집에서 가이드라인을 제시해주는 것이 가치도 더 있어 보이고 우리가 투자하는 리소스 대비 괜찮다고 판단함.
진행을 도와주는 것은 고도화 과정에서 추가해도 충분하다고 생각한다

**Reply by woosung1223** [reply]
레벨3에서는 리소스가 상대적으로 부족하니, 마코가 제시한대로 개설 폼을 응용하는 방법이 좋아보입니다!

**Reply by yeopto** [reply]
폼을 이용해서 입력받아서 가이드라인을 제시하는 것 좋은 것 같네요. 그걸 룸에다 띄워주는거죠?

그렇다면 폼으로 받는 기능으로 시작해서 
나중에는 스터디룸에서 함께 공동 작업할 수 있게 고도화하면 좋을 것 같네요.

**Reply by jaehee329** [reply]
스터디 개설 폼에서 짧은 주제에 대한 학습 방법 계획을 입력하게 하자는 말일까?

데모에서 이런 패러다임의 스터디를 해본 사람이 많이 없어서 그 사람들이 좋은 경험을 하게 해 붙잡아둬야 한다는 피드백을 받았는데, 
이 학습 방법을 우리가 제시하지 않고 유저에게 자유롭게 입력하게 하면 오히려 사용을 어렵게 만드는 요인이 되지 않아 걱정돼

**Reply by MoonJeWoong** [reply]
최소한의 기능으로 모집 프로세스 구현 이후, 효능감을 위한 템플릿을 유저가 우리 서비스 상에서 스터디 과정 중 작성해서 각자 업로드할 수 있는 기능 정도로 구현하면 어떨지?

**Reply by woo-jk** [reply]
툴 제공보다는 가이드 라인 제공이 가성비가 엄청 좋긴할듯 !

**Comment by MoonJeWoong** [comment]
1. 일단 하루스터디 서비스 사용자 구체화가 필요하다. 짧고 빠른 스터디를 원하는 사람이 도대체 누구인가? 특정해서 구체화하는 것이 필요하고 이를 기반으로 다시 풀어내야할 문제를 정의해야 한다.
  - 예시 : 우테코나, 회사 내와 같은 특정 그룹을 대상으로 짧고 빠른 스터디를 제공하는 서비스라고 특정하면 어떨까??
2. 그 다음 최소한의 MVP 재설정
3. 짧고 빠른 스터디에서 사용자가 최대한의 효능감을 얻어내기 위한 템플릿 제공

**Reply by aak2075** [reply]
1. 사용자 구체화를 지금 하는게 맞을지 조심스럽다. 어쨌든간에 우리의 사용자는 짧고 빠른 스터디를 원하는 사람인것이고. 문제는 이러한 짧은 스터디 경험을 제공할 수 있는지가 아닐까?

2. 최소한의 MVP 재설정 좋아요👍

3. 템플릿 제공도 동의~

**Reply by yeopto** [reply]
2번, 3번 동의~

사용자가 빠르게 효과적으로 지식을 습득하고 싶은 사람들이란건 변하지 않는다 생각해요.

짧고 빠르게에서 한정짓지말고
짧고 빠르게 스터디하여 반복적인 성취를 느껴 효능감을 주는 것으로 우리의 서비스 가치에 초점을 맞추는게 좋아보여요.

그렇게 효능감을 느낄 수 있는 방법을 제시해주는 서비스로 방향성을 잡는게 좋다고 생각해서 처음엔 템플릿으로 시작하는게 좋은 아이디어인거같네요

**Reply by jaehee329** [reply]
2, 3에 대해서는 완전 동의야
사용자 구체화가 우선순위 1순위인 이유를 더 설명해줄 수 있을까?

현재 상황에선 기존 설정한 요구사항에서 기능을 축소하는 것은 거의 확실시되는데,
시간이 한정된 상황에서 사용자 구체화부터 다시 논의하는 것이 
리소스 대비 MVP를 재설정하고 템플릿을 제공하는 것에 도움이 많이 될 지 의문이야.

**Comment by nlom0218** [comment]
### 2차 데모까지의 목표

우리가 처음에 설정했던 MVP에서 프로필 페이지 제외한 것들

### 앞으로의 방향성

스터디가 어떻게 하면 짧은 시간에 최대 효율을 낼 수 있고, 효율적으로 진행될 수 있는 방법에 대해 살펴본다. 이를 통해 우리가 제시하는 스터디 방법을 하루스터디에서 경험을 할 수 있도록 제공한다.

1. 모집할 때, 어떤 방식으로 스터디가 진행될지에 대한 선택을 스터디장이 정할 수 있다.
2. 스터디 룸에서 내가 선택한 방법에 따른 스터디 진행 탭이 생긴다. 해당 탭에서 스터디를 진행할 때 필요한 툴을 제공한다.
3. 회고 탭에서 스터디에 대한 여러 회고를 진행할 수 있도록 돕는다.

위와 같은 과정을 통해 진행중, 완료 된 시점에서 스터디가 꾸준히 진행되고 관리되고 있음을 사용자가 느낄 수 있게 한다. 스터디에 대한 회고는 자신만의 블로그도 될 수 있다.

**Reply by aak2075** [reply]
한 가지의 스터디 방식만 제공하는 것이 아니라 여러 방식을 제공하고 선택하는 것도 좋은 의견이네요👍

2번도 공통적으로 나와서 같이 얘기해보면 좋을 것 같다.

3번은 이해가 잘 안가서 조금 더 설명해줄 수 있을까요?

**Reply by MoonJeWoong** [reply]
앞에서 얘기한 것처럼 스터디를 진행할 때 효능감을 최대화 시킬 수 있는 툴(템플릿)을 조사하고 연구하는 과정에 리소스가 많이 투입되어야 할 것 같아용. 기능적인 부분을 잘 쳐내고 일정 추산을 잘 해보고 진행해봐야 할 것 같습니다.

**Comment by woosung1223** [comment]
**1. 현재 MVP 기능 최소화**
  - 뱃지 기능, 검색 기능은 반드시 필요할까?
  - 모집 기능을 제외한 기능은 당장은 구현하지 않는다.


**2. 하루스터디 서비스 접근성을 낮출 필요가 있음**
  - 기존 스터디 패러다임과는 다름
  - 따라서 불안감, 걱정을 느끼지 않도록 시스템을 구성해야 함


**3. 사용자 효능감 극대화 방법 논의 필요**
  - 스터디를 진행하고 있는 스터디원들을 서포팅해줄 장치들이 필요함
    - ChatGPT를 통해 퀴즈를 자동 생성한다거나
    - 스터디 시 사용할 자료를 지정된 시간까지 스터디원들에게 제공받아야 한다거나


**4. 최대한 유저가 스터디 관리에 관여하는 부분을 줄일 필요가 있음**
  -  스터디장이나 스터디원은 단순히 '학습'에만 집중할 수 있게 장치 마련

**Reply by nlom0218** [reply]
나도 당장은 검색은 필요 없을 거 같다고 생각해. 빠르게 모집된 스터디를 찾는 것에 집중을 한다면 점점 고도화를 할 수 있지만 현재는 효능감에 대해 집중하는 것이 좋다고 생각!!

사용자 효능감 극대화 방법에 대해 여러 자료를 찾아보고 몇가지를 선정하는 것이 좋을듯!

**Reply by aak2075** [reply]
1. 뱃지를 제외하는 것만 생각했는데 검색도 고려해보면 좋긴 할듯. 일단 모집 기능만 갖춘 서비스를 출시하고 피드백 받아서 필요하다면 추가해도 좋을듯.

🚨효능감 극대화 방법 논의 중요🚨

**Reply by MoonJeWoong** [reply]
나도 검색을 mvp에서 제외하는거 동의. 진짜 순수하게 모집 기능만 생각한다면 공고글 create만 하고 해당 글의 댓글로 템플릿 공유해서 스터디 참여 과정에서 효능감을 끌어올릴 수 있는 양식을 작성해 공유할 수 있도록 하는 방향도 좋다고 생각함.

**Reply by yeopto** [reply]
MVP 다시 생각해보니 검색은 필요없을 것 같긴하네요. 그냥 리스트 한번에 뿌려주는 방향으로가도 나쁘지 않을 것 같아요.

gpt api를 이용해서 키워드로 문제 만들어달라해서 그걸 종료전에 맞춰야 스터디완료하는 플로우를 가져가도 좋을 것 같네요.

**Comment by woo-jk** [comment]
### 두 가지 방향성
1. 우리가 지금까지 회의하고 만들어 온 것은 '모집 플랫폼'
2. 오늘 데모데이와 준의 조언을 통해 조언 받은 것은 '스터디 진행 도구'

두 방향성을 모두 안고가면 좋겠지만 개인적인 생각으론 두 기능 다 챙기다간 둘 다 흐지부지 될 것 같다고 생각함.. 한 가지 방향성을 확실하게 선택하면 좋을 듯 

### 내 생각
- 지금까지 해온 것이 아깝긴 하지만 '진행 도구'에 초점을 맞춰 방향성을 재설정하고, '모집 플랫폼'은 잠시 넣어 두는건 어떨까?(버리자는 의미 아님!) 
- 진행 도구가 만족스럽게 개발 되면 그 때 모집 플랫폼을 다시 꺼내서 구현하면 좋을 것 같다고 생각함

**Reply by jaehee329** [reply]
진행 도구 개발이 기존에 기획하던 모집 플랫폼보다 선행된다면 다시 초기 기획단계로 돌아가야 할 것 같아.
현실적인 문제로 모집 플랫폼을 먼저 하고 이후 툴을 만들어 사용을 유도하는 방향이 어떨까 하는데, 

도구가 먼저 개발된다면 어떻게 다음 데모까지 일정 추정을 해야 할까?

**Reply by woosung1223** [reply]
전 개인적으로 모집 플랫폼 정체성은 '최소한'으로라도 유지해야 한다고 생각해요.

그리고 조금 현실적으로 생각하면 진행 도구쪽도 되게 고도화 할 내용이 많은 것 같은데,
API 설계까지 진행되고 있는 지금 돌아가기에는 부담스러운 면이 없지 않아 있네요 🥲

**Reply by nlom0218** [reply]
모디의 의견과 많이 이어지는 거 같어! 거기에 내 생각 적었으니 보고 참고!ㅎㅎ

**Reply by MoonJeWoong** [reply]
위에서도 적은 내용이지만 모집 기능을 정말 최소한으로 한다면 공고글만 올리면 스터디 진행이 가능하고 해당 페이지에서 스터디 자료를 공유할 수 있는 댓글 기능으로 제공한다고 생각하면 가능할 것 같습니다.

**Reply by yeopto** [reply]
스프린트 특정상 지금 새로운 방향으로 트는 것은 무모할 것 같아요.

최대한 간단하게 해서 어떻게 효능감을 주는게 뭘지 고민하는게 최우선일거같아요.

**Reply by aak2075** [reply]
진행 도구에 초점을 맞춰도 물론 좋지만 백엔드는 기술적으로 챌린지가 없거나 너무 어렵거나 일 것 같아요.

다른 부분은 공통된 의견이 있으니 같이 얘기해보자.

**Reply by woo-jk** [reply]
방향성 아예 바꾸자는 말 취소~

**Comment by jaehee329** [comment]
다들 갑작스럽게 혼란스러운 상황에서 이 회의가
일단 2차 데모까지 어떻게 목표를 잡을지 정도로 우리의 생각을 일치시켜서 서로 안심하는 방향으로 갔으면 좋겠어요~
